bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1796,43 +1858,111 @@ OperationResult PreProcessor::launchReqPreProcessing(const string &batchCid,
                                << GlobalData::current_block_id << "] delta [" << GlobalData::block_delta << "]");
     GlobalData::increment_step = true;
   }
-  auto preProcessResultBuffer = (char *)getPreProcessResultBuffer(clientId, reqSeqNum, reqOffsetInBatch);
-  IRequestsHandler::ExecutionRequest request = bftEngine::IRequestsHandler::ExecutionRequest{
-      clientId,
-      reqSeqNum,
-      reqCid,
-      PRE_PROCESS_FLAG,
-      preProcessReqMsg->requestLength(),
-      preProcessReqMsg->requestBuf(),
-      std::string(preProcessReqMsg->requestSignature(), preProcessReqMsg->requestSignatureLength()),
-      maxPreExecResultSize_,
-      preProcessResultBuffer,
-      reqSeqNum,
-      reqOffsetInBatch,
-      preProcessReqMsg->result()};
-  requestsHandler_.preExecute(request, std::nullopt, reqCid, span);
-  auto preProcessResult = static_cast<OperationResult>(request.outExecutionStatus);
-  resultLen = request.outActualReplySize;
-  if (preProcessResult != OperationResult::SUCCESS) {
+  auto resultBufferEntry = std::ref(getPreProcessResultBufferEntry(clientId, reqSeqNum, reqOffsetInBatch));
```

Comment:
So, the last parameter is always '0' for the first time? That means we almost always will have two trials. Consider creating an adaptive mechanism for the buffer size choice based on previous results.